utils/IrodsConnector.py

```diff
@@ -286,49 +283,81 @@ def _get_session(self):
-                except Exception as sslerror:
-                    raise sslerror
-            except Exception as autherror:
-                logging.info('AUTHENTICATION ERROR', exc_info=True)
-                print(f'{RED}AUTHENTICATION ERROR: {autherror!r}{DEFAULT}')
-                raise autherror
+                except irods.connection.PlainTextPAMPasswordError as ptppe:
+                    print(f'{RED}ENVIRONMENT INCOMPLETE? {ptppe!r}{DEFAULT}')
+                    try:
+                        ssl_context = ssl.create_default_context(
+                            purpose=ssl.Purpose.SERVER_AUTH,
+                            cafile=None, capath=None, cadata=None)
+                        ssl_settings = {
+                            'client_server_negotiation':
+                                'request_server_negotiation',
+                            'client_server_policy': 'CS_NEG_REQUIRE',
+                            'encryption_algorithm': 'AES-256-CBC',
+                            'encryption_key_size': 32,
+                            'encryption_num_hash_rounds': 16,
+                            'encryption_salt_size': 8,
+                            'ssl_context': ssl_context,
+                        }
+                        options.update(ssl_settings)
+                        print('PARTIAL ENVIRONMENT SESSION')
+                        self._session = irods.session.iRODSSession(**options)
+                    except Exception as error:
+                        print(f'{RED}PARTIAL ENVIRONMENT LOGIN FAILED: {error!r}{DEFAULT}')
+                        raise error
+                except Exception as autherror:
+                    logging.info('AUTHENTICATION ERROR', exc_info=True)
+                    print(f'{RED}AUTHENTICATION ERROR: {autherror!r}{DEFAULT}')
+                    raise autherror
+            try:
+                # Check for good authentication and cache PAM password
+                if 'password' in options:
+                    _ = self._session.pool.get_connection()
+                    pam_passwords = self._session.pam_pw_negotiated
+                    if len(pam_passwords):
+                        with open(irods_auth_file, 'w', encoding='utf-8') as authfd:
+                            authfd.write(
+                                irods.password_obfuscation.encode(pam_passwords[0]))
+            except (irods.exception.CAT_INVALID_AUTHENTICATION, KeyError) as error:
+                raise error
             print('Welcome to iRODS:')
             print(f'iRODS Zone: {self._session.zone}')
             print(f'You are: {self._session.username}')
             print(f'Default resource: {self.default_resc}')
             print('You have access to: \n')
             home_path = f'/{self._session.zone}/home'
-            colls = self.get_collection(home_path).subcollections
-            print('\n'.join([coll.path for coll in colls]))
+            if self._session.collections.exists(home_path):
```

Comment:
This is a legacy operation and will affect only a small number of users.  I'm not sure how to improve it beyond removing it.  This will likely happen when the script becomes a proper program and there is no more console view, except maybe for command-line users.  Also, whenever `/<zone>/home` is accessed in one of the views, it will also be an issue like it already is for the web GUI.